src/Bootstrap.php

```diff
@@ -3,79 +3,212 @@
+        $game = new self();
+        return $game->getGame();
+    }
+    
+    public function getGame()
+    {
+        $config = $this->createConfiguration();
+        $logger = $this->createLogger($config, "lotgd");
+        $composer = $this->createComposer($logger);
+        $this->bootstrapClasses = $this->getBootstrapClasses($composer);
+        
+        $pdo = $this->connectToDatabase($config);
+        $entityManager = $this->createEntityManager($pdo);
+        
+        $eventManager = $this->createEventManager($entityManager);
+        
+        $this->game = new Game($config, $logger, $entityManager, $eventManager);
+        
+        return $this->game;
+    }
+    
+    protected function createEntityManager(\PDO $pdo): EntityManagerInterface
+    {
+        $this->annotationDirectories = $this->generateAnnotationDirectories();
+        $configuration = Setup::createAnnotationMetadataConfiguration($this->annotationDirectories, true);
 
         // Set a quote
         $configuration->setQuoteStrategy(new AnsiQuoteStrategy());
 
+        // Create entity manager
         $entityManager = EntityManager::create(["pdo" => $pdo], $configuration);
 
-        // Create Schema
+        // Create Schema and update database if needed
         $metaData = $entityManager->getMetadataFactory()->getAllMetadata();
         $schemaTool = new SchemaTool($entityManager);
         $schemaTool->updateSchema($metaData);
+        
+        return $entityManager;
+    }
+    
+    /**
+     * Connects to a database using pdo
+     * @param \LotGD\Core\Configuration $config
+     * @return \PDO
+     */
+    protected function connectToDatabase(Configuration $config): \PDO
+    {
+        return new \PDO($config->getDatabaseDSN(), $config->getDatabaseUser(), $config->getDatabasePassword());
+    }
+    
+    /**
+     * Creates and returns an instance of ComposerManager
+     * @param Logger $logger
+     * @return ComposerManager
+     */
+    protected function createComposer(Logger $logger): ComposerManager
+    {
+        $composer = new ComposerManager($logger);
+        
+        return $composer;
+    }
+    
+    /**
+     * Returns all bootstrap classes
+     * @param ComposerManager $composer
+     * @return array
+     * @throws \Exception
+     */
+    protected function getBootstrapClasses(ComposerManager $composer): array
+    {
+        $packages = $composer->getPackages();
+        $classes = [];
+        
+        foreach ($packages as $package) {
+            if (isset($package->getExtra()["lotgd-namespace"]) === false) {
+                continue;
+            }
+            
+            $cn = $package->getExtra()["lotgd-namespace"] . "Bootstrap";
+            
+            // silently ignore that class does not exist, could be one that doesn't need to bootstrap
+            if (class_exists($cn, true) === false) {
+                continue;
+            }
+            
+            $cl = new $cn();
```

Comment:
This seems dangerous. I wouldnt create an instance of it. Who knows what side effects the developer has added to its constructor :)